cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,175 @@
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def test_proxy_invalid(appliance, provider, request):
+    """Configures the specific invalid proxy for each provider
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test Invalid Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], False)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+
+def test_proxy_remove(appliance, provider, request):
+    """Removes the specific valid proxy for all providers
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test to Remove Proxy")
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    appliance.ssh_client.run_command(
+        "rm -f /var/www/miq/vmdb/config/settings.local.yml")
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def test_proxy_override(appliance, provider, request):
+    """ Configures a bad default and then a valid proxy for each provider
+        This one is a little weird because all the other tests setup a specific provider, but
+        this one needs to setup the default incorrectly and the provider correctly so that we know
+        the provider proxy was definitely being used and not the default.
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test Override Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, ['default', provider.type], provider)
+
+    logger.info("Test to fail first")
+    settings_local(appliance, ["default"], False)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+    logger.info("Then test to pass using provider specific override")
+    settings_local(appliance, ['default', provider.type], True)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
```

Comment:
The functions below should not take an appliance argument, moreover they should be methods on the appliance object. They provide functionality that someone else may want to make use of and therefore should not but hidden in a test file.Whilst it is true that the appliance object file is getting horrendously big and needs to be split up, putting these functions here is still against the guidelines and should be moved into the appliance object.